lib/katello/tasks/pulp3_content_switchover.rake

```diff
@@ -0,0 +1,19 @@
+require File.expand_path("../engine", File.dirname(__FILE__))
+
+namespace :katello do
+  desc "Runs a Pulp 3 migration of pulp3 hrefs to pulp ids for supported content types."
+  task :pulp3_content_switchover => :environment do
+    content_models = Katello::Pulp3::Migration::CONTENT_TYPES
+
+    content_models.each do |model|
+      if model.where(migrated_pulp3_href: nil).any?
+        $stderr.print("ERROR: at least one #{model.table_name} record has migrated_pulp3_href NULL value")
+        exit 1
+      end
+    end
+
+    content_models.each do |model|
+      model.update_all("pulp_id = migrated_pulp3_href")
```

Comment:
Wonder if something like `model.where.not("pulp_id=migrated_pulp3_href").update_all("pulp_id=migrated_pulp3_href")` will make this more efficient on re-runs or cause memory overloads in case of large number of records like packages?

Reply:
A good point. I wasn't sure about every context it would be run in, and I don't see a reason to not add this clause. Unless @jlsherrill has an objection, I'll add it. Thanks!

Reply:
that seems reasonable!